project-keeper/src/test/java/com/exasol/projectkeeper/validators/changesfile/ChangesFileIOTest.java

```diff
@@ -24,37 +25,142 @@ void testParsing() throws IOException {
         final ChangesFile changesFile = new ChangesFileIO().read(changesFilePath);
         final List<String> headings = changesFile.getSections().stream().map(ChangesFileSection::getHeading)
                 .collect(Collectors.toList());
-        assertThat(changesFile.getHeading(), equalTo("# My Project 0.1.0, released 1980-01-01"));
-        assertThat(headings, contains("## Summary", "## Features", "## Bug Fixes", "## Documentation", "## Refactoring",
-                "## Dependency Updates"));
+        assertThat(changesFile.getProjectName(), equalTo("My Project"));
+        assertThat(changesFile.getProjectVersion().toString(), equalTo("0.1.0"));
+        assertThat(changesFile.getReleaseDate(), equalTo("1980-01-01"));
+        assertThat(changesFile.getParsedReleaseDate().get(), equalTo(LocalDate.parse("1980-01-01")));
+        assertThat(changesFile.getSummarySection().get().getContent(), contains("", "My summary", ""));
+        assertThat(headings, contains("## Features", "## Bug Fixes", "## Documentation", "## Refactoring"));
+        assertThat(changesFile.getDependencyChangeSection().get().getHeading(), equalTo("## Dependency Updates"));
+        assertThat(changesFile.getDependencyChangeSection().get().getContent(), hasSize(18));
     }
 
     @Test
     void testWriting() throws IOException {
-        final ChangesFile changesFile = ChangesFile.builder().setHeader(List.of("# MyChanges"))
-                .addSection(List.of("## My Subsection")).build();
+        final ChangesFile changesFile = ChangesFile.builder().projectName("project").projectVersion("1.2.3")
+                .releaseDate("2023-??-??").codeName("my code name")
+                .summary(ChangesFileSection.builder("## Summary").addLine("my summary content").build())
+                .addSection(ChangesFileSection.builder("# MyChanges").build())
+                .addSection(ChangesFileSection.builder("## My Subsection").addLine("content").build()).build();
         final Path testFile = this.tempDir.resolve("myFile.md");
         new ChangesFileIO().write(changesFile, testFile);
-        assertThat(Files.readString(testFile),
-                equalTo("# MyChanges" + System.lineSeparator() + "## My Subsection" + System.lineSeparator()));
+        assertThat(Files.readString(testFile), equalTo(
+                "# project 1.2.3, released 2023-??-??\n\nCode name: my code name\n\n## Summary\nmy summary content\n# MyChanges\n## My Subsection\ncontent\n"));
```

Comment:
Actually I would prefer the `System.lineSeparator()` approach.
Or is maybe `\n` converted silently?